ansible/roles_ocp_workloads/ocp4_workload_lpe_3scale/files/3scale_single_tenant.yml

```diff
@@ -0,0 +1,122 @@
+# The following is only appropriate when create generic tenants whose names are sequential
+# The following is skipped when creating a "named" tenant, ie; openbanking-dev  
+
+# Name of 3scale API administrator of the tenant
+- set_fact:
+    ocp4_workload_lpe_3scale_tenant_admin_id: "{{ ocp4_workload_lpe_3scale_tenant_admin_name_base }}"
+
+# Name of 3scale tenant organisation
+- set_fact:
+    ocp4_workload_lpe_3scale_org_name: "{{ocp4_workload_lpe_3scale_tenant_user_name_base}}-tenant"
+
+# Name of 3scale tenant namespace
+- set_fact:
+    ocp4_workload_lpe_3scale_tenant_gw_project: "{{ ocp4_workload_lpe_3scale_tenant_user_name_base }}-gw"
+
+# Name of 3scale tenant namespace display name
+- set_fact:
+    ocp4_workload_lpe_3scale_tenant_gw_projectDisplay: "{{ ocp4_workload_lpe_3scale_tenant_user_name_base }} APICast Gateways Project"
+
+# Set the tenant admin email
+- name: "{{ocp4_workload_lpe_3scale_org_name}}      1)  **********   TENANT CREATION  **********"
+  set_fact: 
+    ocp4_workload_lpe_3scale_tenant_admin_email: "{{ocp4_workload_lpe_3scale_admin_email_user}}@{{ocp4_workload_lpe_3scale_admin_email_domain}}"
+- debug: 
+    msg: "{{ocp4_workload_lpe_3scale_org_name}} ocp4_workload_lpe_3scale_tenant_admin_email = {{ocp4_workload_lpe_3scale_tenant_admin_email}}"
+
+# Wait for 3scale tenant to be created
+- name: "Create the {{ocp4_workload_lpe_3scale_org_name}} CRDs"
+  k8s:
+    state: present
+    merge_type:
+    - strategic-merge
+    - merge
+    definition: "{{ lookup('template', item ) | from_yaml }}"
+  loop:
+  - ./templates/tenant-admin-secret.yml.j2
+  - ./templates/tenant.yml.j2
+
+# Routes are created after some delay. Wait here for routes to  be ready.
+- pause:
+    minutes: 1
+
+# Get the {{ ocp4_workload_lpe_3scale_org_name}} Admin URL
+- name: Get the {{ ocp4_workload_lpe_3scale_org_name}} Admin URL
+  shell: >
+    oc get secret "{{ ocp4_workload_lpe_3scale_org_name }}-tenant-secret"  -o json -n "{{ ocp4_workload_lpe_3scale_namespace }}" | jq -r .data.adminURL | base64 -d
+  register: tenant_admin_url_reg
+  ignore_errors: True
+
+# Get the {{ ocp4_workload_lpe_3scale_org_name}} Access Token
+- name: Get the {{ ocp4_workload_lpe_3scale_org_name}} Access Token
+  shell: >
+    oc get secret "{{ ocp4_workload_lpe_3scale_org_name }}-tenant-secret"  -o json -n "{{ ocp4_workload_lpe_3scale_namespace }}" | jq -r .data.token | base64 -d
+  register: tenant_access_token_reg
+  ignore_errors: True
+
+- set_fact:
+    tenant_admin_url: '{{tenant_admin_url_reg["stdout"]}}'
+
+- set_fact:
+    tenant_access_token: '{{tenant_access_token_reg["stdout"]}}'
+
+# create the tenant Gateway
+- block:
+    - name: "**********  {{  ocp4_workload_lpe_3scale_tenant_gw_project }} TENANT API GW  Provisioning **********"
+      command: "oc adm new-project {{ ocp4_workload_lpe_3scale_tenant_gw_project }} --description='{{ ocp4_workload_lpe_3scale_tenant_gw_projectDisplay }}'"
```

Comment:
Please replace with `kubernetes.core.k8s` - you can either create a `Namespace` or a `ProjectRequest`. But since you're using `oc adm` a `Namespace` with the correct annotations for display etc will be preferable.

Reply:
Resolved